Packs/OktaAuth0/Integrations/OktaAuth0EventCollector/OktaAuth0EventCollector.py

```diff
@@ -0,0 +1,292 @@
+import demistomock as demisto
+from CommonServerPython import *
+
+import time
+
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+VENDOR = "okta"
+PRODUCT = "auth0"
+DEFAULT_LIMIT = 1000
+EXPIRED_TOKEN_RANGE = 23 * 60 * 60  # https://auth0.com/docs/secure/tokens/access-tokens/get-access-tokens#renew-access-tokens
+
+
+''' HELPER FUNCTIONS '''
+
+
+def arg_to_strtime(value: Any) -> Optional[str]:
+    if datetime_obj := arg_to_datetime(value):
+        return datetime_obj.strftime(DATE_FORMAT)
+    return None
+
+
+def prepare_query_params(params: dict, last_run: dict = {}) -> dict:
+    """
+    Parses the given inputs into Okta Auth0 Events API expected params format.
+    """
+    query_params = {}
+
+    if not last_run:  # requesting by time query
+        since = arg_to_strtime(params.get('since'))
+        query_params = {
+            "q": last_run.get('query') or f"date:[{since} TO *]",
+            "sort": "date:1",
+            "per_page": 100
+        }
+
+    else:  # requesting by log_id
+        query_params = {
+            "from": last_run.get('last_id'),
+            "sort": "date:1",
+            "take": 100
+        }
+
+    return query_params
+
+
+def should_refresh_access_token() -> bool:
+    """Check whether the access token in the integration context has expired.
+
+    Returns:
+        bool: Whether the token has expired and should refresh the token.
+    """
+    int_context = get_integration_context()
+    if 'expired_token_time' not in int_context or \
+            int(time.time()) - int_context.get('expired_token_time', 1) > EXPIRED_TOKEN_RANGE:
+        return True
+    return False
```

Comment:
```suggestion
def get_access_token_from_context() -> Optional[str]:
    """Returns an access token if exists in the integration context and is not expired. Otherwise, returns None.

    Returns:
        Optional[str]: A valid access token, or `None`.
    """
    int_context = get_integration_context()
    if 'token_creation_time' not in int_context or \
            int(time.time()) - int_context.get('token_creation_time', 1) > TOKEN_TIME_TO_EXPIRE:
        return None
    return int_context["access_token"]
```
see next comment as well

Reply:
done